### DISCUSSION TITLE
GDR and gdrcopy releationship.-

### DISCUSSION THREAD

**Discussion by changchengx** [NVIDIA/gdrcopy#227] [Q&A]
1. ```GDR``` stands for ```GPUDirect RDMA``` right?
2. Does ```gdrcopy``` library only for ```create user-space mappings of GPU memory``` ? Is it only supporting copying data between CPU and GPU on the same server machine?
3. If it aims to send/recv data between GPUs on the different server machines with using ```GPUDirect RDMA```, does it need ```gdrcopy``` ?

**Comment by jdinan** [comment]
The naming is a little confusing, GDR Copy and GPUDirect RDMA are separate data movement mechanisms. GDR Copy enables the CPU to create a mapping for memory of a GPU on the same server. Using this mapping, the CPU can access the GPU's memory using direct load/store operations. GPUDirect RDMA allows a network controller to directly access the GPU's memory. GDR Copy is not required to use GPUDirect RDMA.

**Comment by lcebaman** [comment]
Yet, higher BW is achieved when GDRcopy and GPUDirect are enabled
